bsdlib/README.rst

```diff
@@ -3,9 +3,131 @@
 BSD Library
 ###########
 
+The BSD library is Nordic Semiconductor's implementation of the BSD Socket API, which is a set of standard function calls that can be used in an application.
+This library aims to be compatible with the BSD socket standard to simplify writing and porting of existing applications. 
+
+The library has the following set of sockets:
+
+* 8 generic sockets that can be used generically between UDP, TCP, TLS, DTLS, and AT commands.
+* 1 GNSS/AGPS socket
+* 1 Modem DFU socket
+* 10 PDN sockets
+
+Overview
+********
+
+.. figure:: doc/images/bsd_lib_overview.svg
+   :alt: BSD library overview diagram
+
+   BSD library overview diagram
+
+The BSD library is the primary interface to operate the nRF9160 modem in order to establish LTE-M, NBIOT, and GNSS connections.
+The library abstracts away the details on memory management and synchronization leaving the application to focus on writing the application using standardized API calls.
+
+The current implementation of the BSD library supports the following protocols:
+
+* TCP (client/server)
+* UDP
+* RAW sockets
+* DTLS 1.2 client
+* TLS 1.2 client
+* AT commands
+* GNSS
+* AGPS
+* PDN management
+* Modem DFU
+
+.. note::
+   The DNS address management protocol is not yet implemented.
+
+The library implements the following standard BSD functions:
+   
+* socket
+* close
+* bind
+* listen
+* accept
+* connect
+* recvfrom
+* recv
+* read
+* sendto
+* send
+* write
+* getsockopt
+* setsockopt
+* fcntl
+* poll
+* inet_pton
+* inet_ntop
+* getaddrinfo
+* freeaddrinfo
+
+.. note::
+   The ``select`` function is not yet implemented.
+
+
+Library internals
+*****************
+
+The purpose of the BSD library is to communicate with the nRF9160 modem firmware.
+This is done by sending events and getting responses over the Inter Processor Communication peripheral (IPC).
+To facilitate data transfer between the application and the modem, payload data is set in the shared RAM.
+
+The modem contains a full IP and DTLS/TLS stack as well as GNSS.
```

Comment:
Can we add links and/or spelled out versions? I have no idea what GNSS is.

Reply:
postponed